### DISCUSSION TITLE
Ability to set KitQLClient url

### DISCUSSION THREAD

**Discussion by JamieDW** [jycouet/kitql#139] [Q&A] [answered]
Hi, thankyou for this amazing project. It's been brilliant to use so far.

Is it possible to change the KitQLClient URL for certain mutations/queries?

I have a case where I am needing to access 'authenticated' operations which reside in a different graphql schema.
For example my base graphql endpoint is 'site.com/graphql' but I need to change the URL to 'site.com/graphql/admin'

Thankyou again.

**Comment by jycouet** [comment, marked as answer]
Thank you for your feedback.
Today this is not possible to have 2 clients (2 endpoints) at the same time.

I wonder how common is this use case? 🤔
You can do a regular `fetch` to login there? or do you have 50% queries on one endpoint and 50% on the other endpoint?

There is a reason to not merge these 2 endpoints? Something like 👇

```mermaid
graph TD;
    subgraph 2 endpoints
    AA[GraphQL A]-->AC[Your client app];
    AB[GraphQL B]-->AC;
    end

    subgraph one mesh - merging 2 endpoints
    BA[GraphQL A]-->MESH[GraphQL Merged];
    BB[GraphQL B]-->MESH[GraphQL Merged];
    MESH-->BC[Your client app];
    end
```

Let me know, I'm curious 👍

**Reply by JamieDW** [reply]
Hi jycouet, thankyou for the quick response.

I would say it's not the most common case but definitely something to think about.

The reason for the request is I am building an API in parallel with the front end. The GraphQl API package I am using has the functionality to split out the schemas into separate endpoints, I'm then using this functionality to provide different authorisation middleware to different endpoints.

However I have found a workaround which Is better suited now, using a single endpoint 🙂

Thank you again for the package!

**Reply by jycouet** [reply]
No, nice to read.
Have fun 🥳
